File: compress/compress.js
// Ryan and my solution
const compress = (s) => {
  let count = 1
  let current = s[0]
  let result = ""
  
  for (let i=0;i<s.length;i++){
    if(current === s[i+1]){
      count++
    }
    else {
      if (count>1){
       result += count 
      }
      result += current
      current = s[i+1]
      count = 1
    }
  }
  
  return result
  
};

// Pointer solution we came up with

const compress = (s) => {
  let i = 0
  let j = 0
  let result = ""
  
  while (j<s.length+1){
    if(s[j] ===s[i]){
      j++
    }
    else{
      let count = j - i
      if( count > 1){
       result += count 
      }
      result += s[i]
      i = j
      j++
    }
  }
    console.log(result)
    return result
  };

File: Structy_2/2_Arrays_Strings/2_Compress/compress.py
def compress(s):
  result = []
  num = 1
  i = 0
  
  while i <= len(s):
    print('here is i', i)
    if(i == len(s)-1):
      print('length = s')
      if num > 1:
        result.append(str(num))
      result.append(s[i])
      break
    print(s[i])
    print(s[i + 1])
    print(num)
    if s[i] == s[i + 1]:
      num += 1
      i += 1
    else:
      print('here')
      if num > 1:
        result.append(str(num))
      result.append(s[i])
      print(result)
      num = 1
      i += 1
  print(result)
  return "".join(result)